File: sdk/python/config/UserMessageConfig.py
############################################################################
#
#  Copyright 2023 Paphus Solutions Inc.
#
#  Licensed under the Eclipse Public License, Version 1.0 (the "License");
#  you may not use this file except in compliance with the License.
#  You may obtain a copy of the License at
#
#      http://www.eclipse.org/legal/epl-v10.html
#
#  Unless required by applicable law or agreed to in writing, software
#  distributed under the License is distributed on an "AS IS" BASIS,
#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
#  See the License for the specific language governing permissions and
#  limitations under the License.
#
############################################################################
from config.Config import Config
from util.Utils import Writer, Utils


class UserMessageConfig(Config):
    id: str
    creationDate: str
    owner: str
    creator: str
    target: str
    parent: str
    subject: str
    message:str 
    def __init__(self):
        super().__init__()
        self.id = None
        self.creationDate = None
        self.owner = None
        self.creator = None
        self.target = None
        self.parent = None
        self.subject = None
        self.message = None

    def parseXML(self, xml):
        super().parseXML(xml)
        root = Utils.loadXML(xml)
        if(root == None):
            return
        self.id = root.attrib.get("id")
        self.creationDate = root.attrib.get("creationDate")
        self.owner = root.attrib.get("owner")
        self.creator = root.attrib.get("creator")
        self.target = root.attrib.get("target")
        self.parent = root.attrib.get("parent")

        # tag name
        if(root.find("subject") != None):
            self.subject = root.find("subject")
        if(root.find("message") != None):
            self.message = root.find("message")

    def toXML(self):
        writer = Writer("<user-message")
        self.writeCredentials(writer)

        if (self.id != None):
            writer.append(" id=\"" + self.id + "\"")

        if (self.creationDate != None):
            writer.append(" creationDate=\"" + self.creationDate + "\"")

        if (self.owner != None):
            writer.append(" owner=\"" + self.owner + "\"")

        if (self.creator != None):
            writer.append(" creator=\"" + self.creator + "\"")

        if (self.target != None):
            writer.append(" target=\"" + self.target + "\"")

        if (self.parent != None):
            writer.append(" parent=\"" + self.parent + "\"")

        writer.append(">")
        
        if (self.subject != None):
            writer.append("<subject>")
            writer.append(Utils.escapeHTML(self.subject))
            writer.append("</subject>")

        if (self.message != None):
            writer.append("<message>")
            writer.append(Utils.escapeHTML(self.message))
            writer.append("</message>")

        writer.append("</user-message>")
        return writer

File: sdk/nodejs/src/config/UserMessageConfig.ts
/*
*  Copyright 2023 Paphus Solutions Inc.
*
*  Licensed under the Eclipse Public License, Version 1.0 (the "License");
*  you may not use this file except in compliance with the License.
*  You may obtain a copy of the License at
*
*      http://www.eclipse.org/legal/epl-v10.html
*
*  Unless required by applicable law or agreed to in writing, software
*  distributed under the License is distributed on an "AS IS" BASIS,
*  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
*  See the License for the specific language governing permissions and
*  limitations under the License.
*/

import Config from "./Config"
import { XMLReader, XMLWriter } from "../util/Utils"
/**
 * DTO for XML content config.
 */
export default class UserMessageConfig extends Config {
    id?:string
    creationDate?:string
    owner?:string
    creator?:string
    target?:string
    parent?:string
    subject?:string
    message?:string

    parseXML(element: any): void {
        super.parseXML(element)
        let reader = new XMLReader(element)
        this.id = reader.readAttribute('id')
        this.creationDate = reader.readAttribute('creationDate')
        this.owner = reader.readAttribute('owner')
        this.creator = reader.readAttribute('creator')
        this.target = reader.readAttribute('target')
        this.parent = reader.readAttribute('parent')
        //Elements
        this.subject = reader.readElement('subject')
        this.message = reader.readElement('message')
    }

    toXML(): string {
        let writer = new XMLWriter('<user-message')
        this.writeCredentials(writer)
        writer.appendAttribute('id', this.id)
        writer.appendAttribute('creationDate', this.creationDate)
        writer.appendAttribute('owner', this.owner)
        writer.appendAttribute('creator', this.creator)
        writer.appendAttribute('target', this.target)
        writer.appendAttribute('parent', this.parent)
        writer.append('>')
        //Elements
        writer.appendElement('subject', this.subject, true)
        writer.appendElement('message', this.message, true)
        writer.append('</user-message>')
        return writer.toString()

    }
}